credsweeper/filters/value_method_check.py

```diff
@@ -30,6 +30,6 @@ def run(self, line_data: LineData, target: AnalysisTarget) -> bool:
         """
         if line_data.is_well_quoted_value:
             return False
-        if "function" in line_data.value or self.PATTERN.search(line_data.value):
+        if "function" in line_data.value or self.PATTERN.match(line_data.value):
```

Comment:
Is there any FP cases that include `function` keyword in the value?
I think checking `function` keyword by `search()` seems reasonable..

Reply:
actually, there was FN case in random generated credentials like "12#@5(!)fs", so I refactored the pattern and applied ``match`` to use the search from begin of a value

Reply:
Hmm.. still i can get it clearly.
Do you mean there is a random generated credential value which includes "function" keyword in it?
Like as "12#@5(!)function3w@91"?
If then i think `search()` method working good.

Reply:
No, i mean the second part of OR condition in the check. 
The ``function`` check does not use regex.

Reply:
Ah.. I just misunderstood about this. 
But in some case like below, `search()` method still more useful doesn't it?

``` python
myPassword = "this_is_not_password"
```

Reply:
The filter does not return True for the line. Only if it will be 
```python
myPassword = this_is_not_password()
```
or
```bash
myPasswsord=password_function
```
*depends on file

Otherside, your code line means that a string is assigned for variable with name "password" 
So, it may be a password. Even "1234"  may be a password, but CredSweeper skips the sequence.

Reply:
Okay i agree. Let's keep the change then.